File: dart/armstrong-numbers/lib/armstrong_numbers.dart
class ArmstrongNumbers {
  bool isArmstrongNumber(String number) => BigInt.parse(number) == armstrongSum(number);

  BigInt armstrongSum(String number) =>
      number.runes
            .where((rune) => 48 <= rune && rune <= 57)   // ascii digits
            .map((rune) => BigInt.from(rune - 48))       // convert to numbers
            .fold(BigInt.zero, (sum, digit) => sum + digit.pow(number.length));
}

File: lua/armstrong-numbers/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
  local n = number
  local len = math.floor(math.log(n, 10)) + 1
  local sum = 0

  while n > 0 do
    local digit = n % 10
    n = n // 10
    sum = sum + digit ^ len
  end

  return sum == number
end

return ArmstrongNumbers
